src/vs/workbench/browser/parts/titlebar/windowTitle.ts

```diff
@@ -100,6 +102,11 @@ export class WindowTitle extends Disposable {
 			this.activeEditorListeners.add(activeEditor.onDidChangeDirty(() => this.titleUpdater.schedule()));
 			this.activeEditorListeners.add(activeEditor.onDidChangeLabel(() => this.titleUpdater.schedule()));
 		}
+		const editor = this.codeEditorService.getActiveCodeEditor() || this.codeEditorService.getFocusedCodeEditor();
+		if (editor) {
+			this.activeEditorListeners.add(editor.onDidBlurEditorText(() => this.titleUpdater.schedule()));
```

Comment:
I think we should start to make the listeners conditional based on the presence of `focusedView` which is not there by default, because we start to do a lot of unnecessary work in most cases. Can we detect if `focusedView` is in settings and only update the title based on view or editor changes in that case?